packages/pluginutils/src/composable-filters.ts

```diff
@@ -231,8 +273,29 @@ export function exprInterpreter(
         ? code.includes(expr.pattern)
         : expr.pattern.test(code);
     }
+    case 'query': {
+      if (id === undefined) {
+        throw new Error('`id` is required for `Query` expression');
+      }
+      if (!ctx.urlSearchParamsCache) {
+        let [_, queryString = ''] = id.split('?', 2);
+        ctx.urlSearchParamsCache = new URLSearchParams(queryString);
+      }
+      let urlParams = ctx.urlSearchParamsCache;
+      if (typeof expr.pattern === 'boolean') {
+        if (expr.pattern) {
+          return urlParams.has(expr.key);
+        } else {
+          return !urlParams.has(expr.key);
+        }
+      } else if (typeof expr.pattern === 'string') {
+        return urlParams.get(expr.key) === expr.pattern;
+      } else {
+        return expr.pattern.test(urlParams.get(expr.key) ?? '');
+      }
+    }
     default: {
-      throw new Error(`Expression kind ${expr.kind} is not expected.`);
+      throw new Error(`Expression kind ${expr} is not expected.`);
```

Comment:
I think graphite is correct here. I think it should be ``throw new Error(`Expression kind ${(expr as any).kind} is not expected.`);`` or ``throw new Error(`Expression ${JSON(expr)} is not expected.`);``.

Reply:
ts will complain: 
![image](https://github.com/user-attachments/assets/2dd94281-d38b-4ebe-8110-9760e7dbb298)

Reply:
I use option two